pep/tests/unit/authenticators/credentials-authenticator/service-test.ts

```diff
@@ -0,0 +1,89 @@
+    });
+
+    test('Calculate correct expiration time', function(assert) {
+        const authenticator = this.owner.lookup('authenticator:credentials');
+        const expirationTime = authenticator._absolutizeExpirationTime(expiresIn);
+        assert.equal(expirationTime, new Date(new Date().getTime() + expiresIn * 1000).getTime());
+    });
+
+    test('Does not return value if not passed expiration time', function(assert) {
+        const authenticator = this.owner.lookup('authenticator:credentials');
+        const expiresIn = undefined;
+        const expirationTime = authenticator._absolutizeExpirationTime(expiresIn);
+        assert.notOk(expirationTime);
+    });
+
+    test('Restores fails with empty object', async function(assert) {
+        const authenticator = this.owner.lookup('authenticator:credentials');
+        assert.rejects(authenticator.restore({}));
+    });
+
+    test('Restores fails without session ID', async function(assert) {
+        const authenticator = this.owner.lookup('authenticator:credentials');
+        const expirationTime = authenticator._absolutizeExpirationTime(expiresIn);
+        assert.rejects(authenticator.restore({ expiresAt: expirationTime }));
+    });
+
+    test('Restores fails if IsValidLogin is false', async function(assert) {
+        const authenticator = this.owner.lookup('authenticator:credentials');
+        const expirationTime = authenticator._absolutizeExpirationTime(expiresIn);
+        assert.rejects(authenticator.restore({ expiresAt: expirationTime, IsValidLogin: false }));
+    });
+
+    test('Restores is successful', async function(assert) {
+        const authenticator = this.owner.lookup('authenticator:credentials');
+        const expirationTime = authenticator._absolutizeExpirationTime(expiresIn);
+        const sessionData = { expiresAt: expirationTime, SessionId: '123', IsValidLogon: true };
+        const result = await authenticator.restore(sessionData);
+        assert.deepEqual(sessionData, result);
+    });
+
+    test('Authentication successful', async function(assert) {
+        const authenticator = this.owner.lookup('authenticator:credentials');
+
+        const result = await authenticator.authenticate('Test6', 'Zedra001');
+
+        assert.equal(result.IsValidLogon, true);
+    });
+
+    test('Unauthenticated session cleared after successful auth', async function(assert) {
+        const authenticator = this.owner.lookup('authenticator:credentials');
+        const session = this.owner.lookup('service:pep-session');
+
+        const expirationTime = authenticator._absolutizeExpirationTime(expiresIn);
+        const sessionData = { expiresAt: expirationTime, SessionId: '123', IsValidLogon: true };
+        session.setUnauthenticatedSession(sessionData);
+
+        let unauthenticatedSession = session.getUnauthenticatedSession();
+        assert.equal(unauthenticatedSession.SessionId, '123');
+
+        await authenticator.authenticate('Test6', 'Zedra001');
```

Comment:
is it ok for valid site creds, even though its a test account, to be stored in the repo? especially since this a public repo.

cc @adistasio

Reply:
@bakerac4 fwiw, i still think these tests are worthwhile, but ideally we would mock the login endpoint with the response you're expecting, and that way you don't have to give it real creds. And also has the added benefit of being alot faster, and not being dependent on the real API being up/functioning, making this more of a true pure unit test (since you're not also testing the api).

but I realize we are under time constraints here, so maybe thats something for the to-do list.